Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp

```diff
@@ -520,6 +525,21 @@ namespace AZ
                 finalMetalSLStr.insert(startOfShaderPos + startOfShaderTag.length() + 1, structuredBufferTempStructs);
             }
 
+            // optimization off attribute "[[clang::optnone]]" added by SPIRV-Cross makes the render 20 times slower
+            // we drop the attribute here, but calculation precision might suffer causing z-fighting
+#if defined(CARBONATED)
+            constexpr const char* strToFind = "[[clang::optnone]]";
+            const size_t strToFindLen = strlen(strToFind);
+            size_t pos = finalMetalSLStr.find(strToFind);
+            while (pos != AZStd::string::npos)
+            {
+                finalMetalSLStr.erase(pos, strToFindLen);
+                pos = finalMetalSLStr.find(strToFind, pos);
+            }
+#else
```

Comment:
Why prefer the manual removal over the regex?

Reply:
Because it is not a regex. The code is longer, but it does exactly the job needed - remove the substrings from the string.
I understand that they want the regex version, I tested mine, it works, I started regex test just in case.

Reply:
Removed find()-erase() cycle, so now it is more detailed comment & changed -MO command line key. Verified locally.

Reply:
I read up on it, the direct find & replace is faster in general rather than regex :)
But since this is a temp workaround it doesn't matter one way or the other.  Once this is tested and all that, feel free to merge back!

Reply:
Resolved conflicts & merged.